ansible/roles/postldap/defaults/main.yml

```diff
@@ -291,45 +292,63 @@ postldap__postfix__dependent_lookup_tables:
   - name: 'ldap_virtual_alias_maps.cf'
     state: 'present'
     comment: |
-      The virtual_alias_maps setting is used to find the target mailbox
-      (configured in the attribute mail, aka the mailbox or maildrop).
+      The virtual_alias_maps setting is used to find the final delivery address
+      (configured in the attribute mailAddress).
     no_log: '{{ postldap__no_log|d(True) }}'
     connection: '{{ postldap__postfix_ldap_connection }}'
     search_base: '{{ postldap__ldap_people_dn|join(",") }}'
     query_filter: '(&
-                    (objectClass=inetOrgPerson)
-                    (|
-                      (mail=%s)
-                      (mailAlias=%s)
-                    )
-                    (|
-                      (authorizedService=all)
-                      (authorizedService=mail:receive)
-                    )
-                  )'
-    result_attribute: 'mail'
+                     (objectClass=inetOrgPerson)
+                     (|
+                       (mailAddress=%s)
+                       (mailAlternateAddress=%s)
+                     )
+                     (|
+                       (authorizedService=all)
+                       (authorizedService=mail:receive)
+                     )
+                   )'
+    result_attribute: 'mailAddress'
+
+  - name: 'ldap_virtual_groups_maps.cf'
+    state: 'present'
+    comment: |
+      The virtual_groups_maps setting is used to find the delivery addresses of
+      the members of a group (mailAlias or mailDistributionList) (configured in
+      the attribute mailForwardTo).
+    no_log: '{{ postldap__no_log|d(True) }}'
+    connection: '{{ postldap__postfix_ldap_connection }}'
+    search_base: '{{ postldap__ldap_groups_dn|join(",") }}'
+    query_filter: '(&
+                     (|
+                       (objectClass=mailAlias)
+                       (objectClass=mailDistributionList)
+                     )
+                     (mailAddress=%s)
+                   )'
+    result_attribute: 'mailForwardTo'
```

Comment:
I did not test it enough.

I have an entry like the following one :
```
dn: ou=IT Department,ou=Groups,dc=tblein,dc=eu
objectClass: organizationalUnit
objectClass: mailDistributionList
objectClass: authorizedServiceObject
authorizedService: all
ou: IT Department
mail: itdept@mydomain.net
mailAddress: itdept@mydomain.net
mailForwardTo: admin1@mydomain.net
mailForwardTo: admin2@mydomain.net
```

The lookup table return also the `mailAdress` of the group:

```
# postmap -q itdept@mydomain.net ldap:/etc/postfix/ldap_virtual_alias_maps.cf

itdept@mydomain.net,admin1@mydomain.net,admin2@mydomain.net
```

Therefore, it ended in a delivery error to the itdept@mydomain.net, while the mail is correctly delivered to the other addresses.

@imrejonk any idea ?

Reply:
Mail aliases forward e-mail messages from an e-mail address (`itdept@mydomain.net`) to other e-mail addresses (`admin1@mydomain.net`, `admin2@mydomain.net`).

Mail accounts accept e-mail messages directed to them (`itdept@mydomain.net`).

You cannot combine the two cases together in one Postfix lookup, otherwise you get a loop where mail for `itdept@mydomain.net` is fowarded to `itdept@mydomain.net`. This is not an issue on the LDAP side, but on the Postfix side, mail aliases need to be looked up separately from normal e-mail addresses.